### PR TITLE
gh-112341: Specify the exact file size if offset is non zero in `socket.sendfile`

### PR DESCRIPTION
**Pull request by aisk**
I think this is a minor fix, so the news entry is not needed.

* Issue: gh-112341

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Is there an easy way to add tests for this?

**Comment by aisk** [comment]
Thanks for the review, test case added.

**Comment by erlend-aasland** [comment]
How does this relate to `os.sendfile`? Will `os.sendfile` and `socket.sendfile` have divergent behaviour? OTOH, how is their behaviour now?

**Comment by erlend-aasland** [comment]
Thanks, @aisk, makes sense to me! cc. @serhiy-storchaka if he want's to take a look. I'm aiming for landing this in a couple of days.

**Comment by erlend-aasland** [comment]
Could you try to compose a short NEWS entry for this? I'm unsure if we really need one; this feels like an implementation detail.

File: Lib/socket.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def _sendfile_use_sendfile(self, file, offset=0, count=None):
             if not fsize:
                 return 0  # empty file
             # Truncate to 1GiB to avoid OverflowError, see bpo-38319.
-            blocksize = min(count or fsize, 2 ** 30)
+            blocksize = min(count or fsize - offset, 2 ** 30)
```
Comment (serhiy-storchaka) [inline Lib/socket.py:363]: What if `offset >= fsize`?

Please add tests for `offset == fsize` and `offset > fsize`.

**Comment by erlend-aasland** [comment]
See also #114077 and #114079.

**Comment by aisk** [comment]
There are some discussions on the original issue #112341. I think this should not be treated as a bug, so I'm closing this. If anyone wants to discuss it further, we can reopen it. Still appreciate the review!

**Pull request closed without merging**